clippy_lints/src/types/redundant_allocation.rs

```diff
@@ -25,19 +25,40 @@ pub(super) fn check(cx: &LateContext<'_>, hir_ty: &hir::Ty<'_>, qpath: &QPath<'_
         }
     }
 
-    if cx.tcx.is_diagnostic_item(sym::Rc, def_id) {
-        if let Some(ty) = is_ty_param_diagnostic_item(cx, qpath, sym::Rc) {
-            let mut applicability = Applicability::MachineApplicable;
-            span_lint_and_sugg(
-                cx,
-                REDUNDANT_ALLOCATION,
-                hir_ty.span,
-                "usage of `Rc<Rc<T>>`",
-                "try",
-                snippet_with_applicability(cx, ty.span, "..", &mut applicability).to_string(),
-                applicability,
-            );
-            true
+    let sym_outer = if cx.tcx.is_diagnostic_item(sym::Rc, def_id) {
+        Some(sym::Rc)
+    } else if cx.tcx.is_diagnostic_item(sym::Arc, def_id) {
+        Some(sym::Arc)
+    } else {
+        None
+    };
+
+    if let Some(sym_outer) = sym_outer {
+        let ty = is_ty_param_diagnostic_item(cx, qpath, sym::Rc)
+            .map_or_else(|| is_ty_param_diagnostic_item(cx, qpath, sym::Arc), |ty| Some(ty));
+
+        if let Some(ty) = ty {
+            if let TyKind::Path(ref qpath_inner) = ty.kind {
+                let inner_span = match get_qpath_generic_tys(qpath_inner).next() {
+                    Some(ty) => ty.span,
+                    None => return false,
+                };
+                let mut applicability = Applicability::MachineApplicable;
+                span_lint_and_sugg(
+                    cx,
+                    REDUNDANT_ALLOCATION,
+                    hir_ty.span,
+                    &format!("usage of `{}<Rc<T>>`", sym_outer),
```

Comment:
The inner type of this lint emission can be `Rc` or `Arc` due to line 38 - 39. Could you adjust the message to reflect the correct type?

Reply:
This one is also implemented :+1: